File: 0649.Dota2Senate.py
class Solution(object):
    def predictPartyVictory(self, senate):
        """
        :type senate: str
        :rtype: str
        """
        queue = collections.deque()
        cnt, ban = [0, 0], [0, 0]
        for ch in senate:
            cnt[ch=='R'] += 1
            queue.append(1 if ch=='R' else 0)
            
        while all(cnt):
            n = queue.popleft()
            if ban[n]:
                ban[n] -= 1
                cnt[n] -= 1
            else:
                ban[n^1] += 1
                queue.append(n)
        return "Radiant" if cnt[1] else "Dire"
        

File: 0000-0999/0649.Dota2Senate.go
func predictPartyVictory(senate string) string {
    var queue []int
    cnt, ban := []int{0, 0}, []int{0, 0}
    for _, s := range senate {
        x := 0
        if s == 'R' {
            x = 1
        }
        cnt[x]++
        queue = append(queue, x)
    }
    for cnt[0] > 0 && cnt[1] > 0 {
        x := queue[0]
        queue = queue[1:]
        if ban[x] > 0 {
            ban[x]--
            cnt[x]--
        } else {
            ban[x^1]++
            queue = append(queue, x)
        }
    }
    if cnt[1] > 0 {
        return "Radiant"
    } else {
        return "Dire"
    }
}